joomla/joomla-cms
# New Feature: Implements a modal to edit the Redirect plugin settings when needed

Pull Request for Issue #7259 

### Summary of Changes
This is a redo of the original done at #7259 with some small code changes

### Testing Instructions
1. Make sure that the Redirect plugin is either disabled or the Collection of URLs is disabled
2. Go to administrator/index.php?option=com_redirect
3. When the collection of URLs is disabled you will see this message:
![image](https://user-images.githubusercontent.com/359377/27507070-1d84f518-58c8-11e7-9173-3e285b2fca27.png)

When the plugin is disabled you will see this message:
![image](https://user-images.githubusercontent.com/359377/27507073-354ff846-58c8-11e7-99fa-81df2930d10b.png)

4. Click on the Redirect System Plugin link and you are taken to a new page where you can edit the plugin settings. Now you need to navigate back to the Redirect plugin.
5. Apply patch
6. When the collection of URLs is disabled you will see this message:
![image](https://user-images.githubusercontent.com/359377/27507101-90497966-58c8-11e7-81de-8120f5fceab2.png)

When the plugin is disabled you will see this message:
![image](https://user-images.githubusercontent.com/359377/27507097-7aa874c2-58c8-11e7-8bb6-1659874677c5.png)

I changed the regular link to a button so it is easier to see where to click.
7. Click on the Redirect System Plugin link and you are shown a modal in which you can change the settings. Update the settings and click Save.
8. The page is refreshed and the message has changed.

### Expected result
I stay on the same page and can edit the settings.

### Actual result
I am taking to a new page and need to navigate back.

### Documentation Changes Required
None

Pinging @brianteeman @franz-wohlkoenig for testing :)
